drivers/bbram/bbram_stm32.c

```diff
@@ -14,7 +14,14 @@
 LOG_MODULE_REGISTER(bbram, CONFIG_BBRAM_LOG_LEVEL);
 
 #define STM32_BKP_REG_BYTES		 4
-#define STM32_BKP_REG_OFFSET		 0x50
+#ifdef TAMP
+/* If a SoC has a TAMP peripherals, then the backup registers are defined there,
+ * not in the RTC.
+ */
+#define STM32_BKP_REG_OFFSET		 (TAMP_BASE + offsetof(TAMP_TypeDef, BKP0R) - RTC_BASE)
+#else
```

Comment:
Is it because the BBRAM base address is given from its parent (RTC) base address resulting in : 
--> BBRAM base address =   TAMP_BASE + offset(of BKP0R)  - RTC_BASE + RTC_BASE

Reply:
Yes. st_stm32_bbram was developped to be a child node to st_stm32_rtc, where bbram would access BKP0R, the first backup register, as an offset from the RTC base address.

This PR adds the _distance_ between RTC and TAMP to this offset, in case bbram is part of the TAMP device.

bbram can then be defined as the rtc's child in the MCU's dts, even if the reference manual has backup registers listed as part of TAMP. My understanding is that RTC and TAMP always run on the same clock domain.